File: ReklamacjeAPI/DTOs/AdminDtos.cs
namespace ReklamacjeAPI.DTOs
{
    public class AdminUserListDto
    {
        public int Id { get; set; }
        public string Login { get; set; }
        public string NazwaWyswietlana { get; set; }
        public string Rola { get; set; }
        public bool IsActive { get; set; }
    }

    public class AdminCreateUserDto
    {
        public string Login { get; set; }
        public string Password { get; set; }
        public string NazwaWyswietlana { get; set; }
        public string Rola { get; set; }
    }

    public class AdminUpdateUserDto
    {
        public string NazwaWyswietlana { get; set; }
        public string Rola { get; set; }
        public bool IsActive { get; set; }
    }

    public class AdminResetPasswordDto
    {
        public string NewPassword { get; set; }
    }
}

File: Ena/app/src/main/java/com/example/ena/api/AdminDtos.java
package com.example.ena.api;

public class AdminDtos {

    // Do pobierania listy (GET /api/admin/users)
    public static class AdminUser {
        public int id;
        public String login;
        public String nazwaWyswietlana;
        public String rola;
        public boolean isActive;
    }

    // Do tworzenia (POST /api/admin/users)
    public static class CreateUserRequest {
        public String login;
        public String password;
        public String nazwaWyswietlana;
        public String rola;

        public CreateUserRequest(String login, String password, String nazwaWyswietlana, String rola) {
            this.login = login;
            this.password = password;
            this.nazwaWyswietlana = nazwaWyswietlana;
            this.rola = rola;
        }
    }

    // Do edycji (PUT /api/admin/users/{id})
    public static class UpdateUserRequest {
        public String nazwaWyswietlana;
        public String rola;
        public boolean isActive;

        public UpdateUserRequest(String nazwaWyswietlana, String rola, boolean isActive) {
            this.nazwaWyswietlana = nazwaWyswietlana;
            this.rola = rola;
            this.isActive = isActive;
        }
    }

    // Do resetu hasła (POST /api/admin/users/{id}/reset-password)
    public static class ResetPasswordRequest {
        public String newPassword;

        public ResetPasswordRequest(String newPassword) {
            this.newPassword = newPassword;
        }
    }
}